test/Microsoft.AspNetCore.SignalR.Client.Tests/HubConnectionTests.cs

```diff
@@ -207,19 +206,21 @@ public class HubConnectionTests
             Assert.Equal("Server timeout (100.00ms) elapsed without receiving a message from the server.", exception.Message);
         }
 
-        [Fact]
-        public async Task OnReceivedAfterTimerDisposedDoesNotThrow()
+        private static Mock<IConnection> CreateMockConnection()
         {
-            var connection = new TestConnection();
-            var hubConnection = new HubConnection(connection, new JsonHubProtocol(), new LoggerFactory());
-            await hubConnection.StartAsync().OrTimeout();
-            await hubConnection.DisposeAsync().OrTimeout();
-
-            // Fire callbacks, they shouldn't fail
-            foreach (var registration in connection.Callbacks)
-            {
-                await registration.InvokeAsync(new byte[0]);
-            }
+            var connection = new Mock<IConnection>();
```

Comment:
/cc @pakrym this is what it looks like to mock a pipe 😄

Reply:
You can implement completed IPipeAwaiter very easy, it would make this mock much cleaner.

Maybe we would end up with a complete mock pipe in a bit.

Reply:
[![https://upload.wikimedia.org/wikipedia/en/thumb/b/b9/MagrittePipe.jpg/300px-MagrittePipe.jpg](https://upload.wikimedia.org/wikipedia/en/thumb/b/b9/MagrittePipe.jpg/300px-MagrittePipe.jpg)](https://en.wikipedia.org/wiki/The_Treachery_of_Images)